source/MaterialXCore/Interface.cpp

```diff
@@ -355,8 +362,29 @@ OutputPtr Input::getConnectedOutput() const
     return result;
 }
 
+InputPtr Input::getInterface() const
```

Comment:
Utility to find the associated input with a given interface name.

Reply:
I understand why adding such a utility simplifies the code elsewhere, it just feels a bit misplaced as a general method on the Input class when it only return interface inputs from nodegraphs, and specifically non-implementation nodegraphs.

One possibility would be to let it return any interface input, both for nodegraphs and nodedefs, and then the caller would need to check which type it is. That would justify having the method here.

But if its _only_ needed specifically for the non-implementation nodegraph case, then maybe it's better to move somewhere else, and renaming it to something more specific, to not confuse API users what the method is doing.

Just my 2 cents, to open a discussion.

Reply:
I think the nodedef check is left over from the code before being extracted and should be removed. Making it a generic method.